sdk/storage/azure_storage_blob/src/clients/page_blob_client.rs

```diff
@@ -0,0 +1,158 @@
+    /// * `container_name` - The name of the container containing this Page blob.
+    /// * `blob_name` - The name of the Page blob to interact with.
+    /// * `credential` - An implementation of [`TokenCredential`] that can provide an Entra ID token to use when authenticating.
+    /// * `options` - Optional configuration for the client.
+    pub fn new(
+        endpoint: &str,
+        container_name: String,
+        blob_name: String,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<PageBlobClientOptions>,
+    ) -> Result<Self> {
+        let mut options = options.unwrap_or_default();
+
+        let storage_headers_policy = Arc::new(StorageHeadersPolicy);
+        options
+            .client_options
+            .per_call_policies
+            .push(storage_headers_policy);
+
+        let oauth_token_policy = BearerTokenCredentialPolicy::new(
+            credential.clone(),
+            ["https://storage.azure.com/.default"],
+        );
+        options
+            .client_options
+            .per_try_policies
+            .push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
+
+        let client = GeneratedPageBlobClient::new(
+            endpoint,
+            credential.clone(),
+            container_name.clone(),
+            blob_name.clone(),
+            Some(options),
+        )?;
+        Ok(Self {
+            endpoint: endpoint.parse()?,
+            client,
+        })
+    }
+
+    /// Gets the endpoint of the Storage account this client is connected to.
+    pub fn endpoint(&self) -> &Url {
+        &self.endpoint
+    }
+
+    /// Gets the container name of the Storage account this client is connected to.
+    pub fn container_name(&self) -> &str {
+        &self.client.container_name
+    }
+
+    /// Gets the blob name of the Storage account this client is connected to.
+    pub fn blob_name(&self) -> &str {
+        &self.client.blob_name
+    }
+
+    /// Creates a new Page blob.
+    ///
+    /// # Arguments
+    ///
+    /// * `content_length` - The maximum size for the Page blob, up to 1TB. The page blob size must
+    ///   be aligned to a 512-byte boundary.
+    /// * `options` - Optional parameters for the request. See [`azure_storage_blob::storage_traits::PageBlobClientCreateOptionsExt`](crate::storage_traits::PageBlobClientCreateOptionsExt) for additional usage helpers.
+    pub async fn create(
+        &self,
+        content_length: u64,
+        options: Option<PageBlobClientCreateOptions<'_>>,
+    ) -> Result<Response<PageBlobClientCreateResult, NoFormat>> {
+        self.client.create(content_length, options).await
+    }
+
+    /// Clears a range of pages.
+    ///
+    /// # Arguments
+    ///
+    /// * `range` - The range of bytes to clear. See [`azure_storage_blob::StorageUtils::format_http_range()`](crate::StorageUtils::format_http_range) for help with the expected String format.
```

Comment:
This is how we plan to ensure discoverability of the unassociated functions. Docs render:
![image](https://github.com/user-attachments/assets/d2999b20-7975-4dee-9fb9-f8e10e0cbf42)

Reply:
Again, don't fully-qualify the whole thing. It's harder to read and unnecessary. It's a link, and clicking on it tells them all they need to know. The rust-analyzer LSP will import/qualify things as needed.